File: app/src/main/java/com/task/libraries/DateWithLabel.java
package com.task.libraries;
import com.task.libraries.DateHelper;
import java.util.Date;

public class DateWithLabel {
    public final String label;
    public final Date date;

    public DateWithLabel(String label, Date date) {
        this.label = label;
        this.date = date;
        if (date == null) {
            throw new IllegalArgumentException("null value provided. " + "label=[" + label + "], date=[" + date + "]");
        }
    }

    @Override
    public String toString() {
        return label;
    }

    @Override
    public int hashCode() {
        throw new IllegalStateException("Not implemented");
    }

    @Override
    public boolean equals(Object o) {
        if (o instanceof DateWithLabel) {
            DateWithLabel newDate = (DateWithLabel) o;
            return label.equals(newDate.label);
        }
        return false;
    }
}

File: app/src/main/java/com/task/libraries/DateWithLabel.kt
package com.task.libraries

import java.util.*

data class DateWithLabel(
    var label: String = "",
    var date: Date? = null
) {
    init {
        date = date?: Date()
    }
   /* init {
        if(date == null)
            throw IllegalArgumentException("null value provided. label=[$label], date=[$date]")
    }
    override fun toString(): String = label
    override fun hashCode() : Int{
        throw IllegalStateException("Not implemented")
    }
    override fun equals(o : Any?): Boolean{
        if(o is DateWithLabel){
            return label == o.label
        }
        return false
    }*/
}